src/client.zig

```diff
@@ -165,14 +154,13 @@ pub const Client = struct {
             .callback = callback,
             .operation = operation,
             .message = message.ref(),
-        }) catch {
-            callback(
+        }) catch |err| switch (err) {
+            .NoSpaceLeft => callback(
                 user_data,
                 operation,
-                empty_result[0..],
-                @enumToInt(ClientError.batch_not_queued),
-            );
-            return;
+                error.BatchNotQueued,
```

Comment:
Maybe a better error here would be `error.TooManyOutstandingRequests` because our request queued has no space left. This error also helps explain why the batch was not queued.